src/mem_table.rs

```diff
@@ -198,42 +245,41 @@ impl KVTable {
     /// Returns None if the key is not in the memtable at all,
     /// Some(None) if the key is in the memtable but has a tombstone value,
     /// Some(Some(value)) if the key is in the memtable with a non-tombstone value.
-    pub(crate) fn get(&self, key: &[u8]) -> Option<ValueWithAttributes> {
-        self.map.get(key).map(|entry| entry.value().clone())
+    pub(crate) fn get(&self, key: &[u8]) -> Option<RowEntry> {
+        let user_key = Bytes::from(key.to_vec());
```

Comment:
Kind of a bummer we have to `to_vec` the key, which (I think) clones it. I wonder if [this one](https://docs.rs/bytes/latest/bytes/struct.Bytes.html#method.from-3) would work? Or I guess Boxing it would still require a clone since ownership would move.. sigh...

Reply:
if i understand correctly, this function also copies the content of bytes. `Bytes` is a lifetime-free type. if we have a reference, it's impossible to drop the lifetime in the type signature without copy 🥲.

it might possible to reduce the copy buy replacing `key: &[u8]` as `key: Bytes` imo.